backend/app/main/main.py

```diff
@@ -290,3 +299,35 @@ def get_session_food_items(food_collect_id, session: Session = Depends(get_db)):
     if not db_session_food_items:
         raise HTTPException(status_code=404, detail="No food_collects")
     return db_session_food_items
+
+
+@router.post("/add_session_food/{food_collect_id}", response_model=int)
+def add_session_food(
+    food_collect_id: int,
+    trays: List[schemas.TrayCreate],
+    food: schemas.FoodCreate,
+    session: Session = Depends(get_db),
+):
+    """
+    Add food and trays in the specified food_collect_id given that the food_collect object has been created
+
+    :return: JSON response with id of the food_collect
+    """
+    db_session_food_collect = crud.get_food_collect(session, food_collect_id)
+    if not db_session_food_collect:
+        raise HTTPException(status_code=404, detail="NO food_collects")
+    else:
+        db_session_food = crud.create_food(
+            session=session, food=food, food_collect_id=food_collect_id
+        )
+        vendor_id = db_session_food_collect.vendor_id
+        for tray in trays:
+            db_session_tray = crud.create_tray(
+                session=session,
+                tray=tray,
+                vendor_id=vendor_id,
+                food_collect_id=food_collect_id,
+            )
+            if not db_session_tray:
```

Comment:
Hey Great Job! One thing to think about is this, if the db_session_tray is somehow not created in the middle of the loop, what happens to the earlier trays that were created? And if this happens many times, will you have remaining excessive trays that end up not being used?

Reply:
@Ty-Won It seems like there's a commit happening after every crud call. Is there a way to rollback beyond commit? Using Session.rollback doesn't seem to do that. 
Also, this may be a naive thinking but, is there a case where crud call fails after commit? It seems like all the crud call here is doing is calling models.Tray() and then committing, so during crud call, any errors at models.Tray() would make the method fail immediately before commit.

Reply:
my mistake, yes there is the commit in the crud call! Nono you're right I think as you mentioned, the rollback won't reverse a whole set of trays if commit has been called after each time. I'm glad you're sharing your thoughts :) its good for discussion. Hmm I don't imagine anything failing after, unless there is one tray incoming from input that is given with improper data types. But this is a really far edge case. Overall looks okay to me :)